src/main/java/duke/data/FileStorage.java

```diff
@@ -0,0 +1,69 @@
+package duke.data;
+
+import duke.entities.Task;
+import duke.entities.Tasklist;
+import duke.entities.Todo;
+
+import java.io.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class FileStorage implements IStorage {
+    private final String filePath;
+    private static final Pattern TASK_REGEX = Pattern.compile("^\\[(T|D|E)\\]\\[(x| )\\] (.*)(?: \\(.*: (.*)\\))?$");

```

Comment:
I like how you abstracted the regex into a constant to easily retrieve the task data in your `load` function 👍